src/platform/Linux/BLEManagerImpl.cpp

```diff
@@ -278,17 +282,14 @@ void BLEManagerImpl::HandlePlatformSpecificBLEEvent(const ChipDeviceEvent * apEv
     case DeviceEventType::kPlatformLinuxBLEPeripheralAdvStartComplete:
         VerifyOrExit(apEvent->Platform.BLEPeripheralAdvStartComplete.mIsSuccess, err = CHIP_ERROR_INCORRECT_STATE);
         sInstance.mFlags.Clear(Flags::kControlOpInProgress).Clear(Flags::kAdvertisingRefreshNeeded);
-
-        if (!sInstance.mFlags.Has(Flags::kAdvertising))
-        {
-            sInstance.mFlags.Set(Flags::kAdvertising);
-        }
-
+        // Start a timer to make sure that the fast advertising is stopped after specified timeout.
+        SuccessOrExit(err = DeviceLayer::SystemLayer().StartTimer(kFastAdvertiseTimeout, HandleAdvertisingTimer, this));
+        sInstance.mFlags.Set(Flags::kAdvertising);
```

Comment:
Would you say it would make sense to add `ExitNow();` here?

This would make the current assumption that is where we end up after this is successfully run, to be more concrete. That way someone doesn't accidentally add some SuccessOrExit in between this successful call to `StartTimer` and `exit` that would possibly, incorrectly, fail to call CancelTimer in that failure case?

Reply:
I don't think that such thing is necessary here. In case of error, the CHIPoBLE service is disabled, which stops advertising (which should generate `kPlatformLinuxBLEPeripheralAdvStopComplete` event, which cancels timer).

Anyway, I've added explicit `CancelTimer()` call in the "error" branch. So, even if the `mBLEAdvertisement.Stop()` fails to generate the event, the timer will be stopped.